ryhope/src/storage/pgsql/epoch_mapper.rs

```diff
@@ -51,21 +56,34 @@ impl EpochMapperStorage {
         let cache = {
             let connection = db.get().await?;
             let mapper_table_name = mapper_table_name(table.as_str());
-            let mut cache = InMemoryEpochMapper::new_empty();
             let rows = connection
                 .query(
                     &format!(
-                        "SELECT {USER_EPOCH}, {INCREMENTAL_EPOCH} FROM {}",
+                        "SELECT {USER_EPOCH}, {INCREMENTAL_EPOCH} FROM {} ORDER BY {USER_EPOCH}",
                         mapper_table_name,
                     ),
                     &[],
                 )
                 .await
                 .context("while fetching incremental epoch")
                 .unwrap();
-            for row in rows {
+            ensure!(
+                !rows.is_empty(),
+                "Loading from empty table {mapper_table_name}"
+            );
+            let read_row = |row: &Row| {
                 let user_epoch = row.get::<_, i64>(0) as UserEpoch;
                 let incremental_epoch = row.get::<_, i64>(1) as IncrementalEpoch;
+                (user_epoch, incremental_epoch)
+            };
+            let (user_epoch, incremental_epoch) = read_row(&rows[0]);
+            ensure!(
+                incremental_epoch == INITIAL_INCREMENTAL_EPOCH,
+                "Wrong initial epoch found in table {mapper_table_name}"
+            );
```

Comment:
Can you read all the rows, _then_ check that the first one is correct? I think the perf. gain will be negligible, but that makes the flow more complex to follow, with a free-standing function and a two-times parsing of DB output.

Reply:
Actually the main reason I am fetching the first row outside of the loop is because I need the initial `user_epoch` to properly initialize the `cache`. The other way I thought would be using an `Option` for the `cache` and initialize it in the first iteration of the loop, but that looked to me more contrived than the current implementation.

Reply:
Ah, right, sorry, I missed that.